src/libraries/System.Private.CoreLib/src/System/Threading/Thread.cs

```diff
@@ -133,7 +133,7 @@ public CultureInfo CurrentUICulture
         {
             get
             {
-                if (s_asyncLocalPrincipal is null)
+                if (s_asyncLocalPrincipal is null || s_asyncLocalPrincipal.Value is null)
```

Comment:
This is going to call `AsyncLocal.Value` twice in the common case - once here, and second time below. Would it be better to call it just once? Something like:

```
IPrincipal? principal = s_asyncLocalPrincipal?.Value;
if (principal == null)
{
    CurrentPrincipal = (principal = AppDomain.CurrentDomain.GetThreadPrincipal());
}
return principal;
```

Reply:
Yep will do